src/main.rs

```diff
@@ -42,29 +42,30 @@ async fn handle_message(
     )
     .await?;
 
-    // Transaction
-    db_adapters::transactions::store_transactions(
+    // Transactions
+    let transactions_future = db_adapters::transactions::store_transactions(
         pool,
         &streamer_message.shards,
-        &streamer_message.block.header.hash.to_string(),
+        &streamer_message.block.header.hash,
         streamer_message.block.header.timestamp,
-    )
-    .await;
+        streamer_message.block.header.height,
+    );
 
     // Receipts
-    for shard in &streamer_message.shards {
-        if let Some(chunk) = &shard.chunk {
-            db_adapters::receipts::store_receipts(
-                pool,
-                &chunk.receipts,
-                &streamer_message.block.header.hash.to_string(),
-                &chunk.header.chunk_hash,
-                streamer_message.block.header.timestamp,
-                strict_mode,
-            )
-            .await?;
-        }
-    }
+    let receipts_future = db_adapters::receipts::store_receipts(
+        pool,
+        &streamer_message.shards,
+        &streamer_message.block.header.hash,
+        streamer_message.block.header.timestamp,
+        strict_mode,
+    );
+
+    // We can process transactions and receipts in parallel
+    // because most of receipts depend on transactions from previous blocks,
+    // so we can save up some time here.
+    // In case of local receipts (they are stored in the same block with corresponding transaction),
+    // we hope retry logic will cover it fine
+    try_join!(transactions_future, receipts_future)?;
```

Comment:
@frol I think we want to put this change in a separate PR and test the performance separately

Reply:
I'd suggest including it here. We can deploy on separate testnet node to check how's the performance and revert it if necessary